### DISCUSSION TITLE
Init exception after migrating to .net8

### DISCUSSION THREAD

**Discussion by mcocaroUX** [unoplatform/uno#15360] [Q&A] [answered]
After migrating from .net7 to 8, app started crashing on the apps init. Specifically on 
`Host = await builder.NavigateAsync<Shell>();`

![image](https://github.com/unoplatform/uno/assets/94198647/44cf92a8-c9f5-4bfa-98d2-9509b7df8cd3)

Stacktrace:
```
16:51:10:590	System.InvalidOperationException: Operation is not valid due to the current state of the object.
16:51:10:590	   at WinRT.ExceptionHelpers.<ThrowExceptionForHR>g__Throw|39_0(Int32 hr)
16:51:10:590	   at WinRT.ExceptionHelpers.ThrowExceptionForHR(Int32 hr)
16:51:10:590	   at ABI.Windows.Storage.IApplicationDataStaticsMethods.get_Current(IObjectReference _obj)
16:51:10:590	   at Windows.Storage.ApplicationData.get_Current()
16:51:10:590	   at Uno.Extensions.Toolkit.ThemeService.GetSavedTheme()
```
I checked and updated all Uno packages but error still ocurring. Any ideas on what could be the issue?

**Comment by jeromelaban** [comment]
@nickrandolph any idea?

**Comment by formalist** [comment]
I'm getting a similar error on a brand new project (.net 8 and latest versions of uno stuff) when doing something like: 

`var s = ApplicationData.Current.LocalSettings.Values["xxx"];`

'ApplicationData.Current' throws the exception 'Operation is not valid due to the current state of the object.'

**Comment by nickrandolph** [comment]
@mcocaroUX is the exception you're seeing just an internal exception (ie application continues if you let it), or does it crash the app? Asking because for unpackaged apps, there are some internal exceptions thrown by storage wrapper that happen because ApplicationData doesn't exist.

**Comment by formalist** [comment]
> because ApplicationData doesn't exist

does this mean that it's no longer possible to save/retrieve application settings, like in https://platform.uno/docs/articles/features/applicationdata.html#storing-settings ?   :-(

**Reply by jeromelaban** [reply]
It's available, but it depends on when in the app you're using it. If you're using it too early, it may fail this way in `.Windows` (winappsdk) head, or in unpackaged apps. It's a windows behavior, it's not something Uno can adjust.

**Reply by formalist** [reply]
I'm using in a model, bound to a page called after a login page, to retrieve some application settings. The application has already started.

**Reply by jeromelaban** [reply, marked as answer]
Then it is related to the fact that the app is running unpackaged: https://github.com/microsoft/WindowsAppSDK/discussions/2618#discussioncomment-6081416. In this case, you can use your own settings system, or use [Uno.Extensions support for settings](https://platform.uno/docs/articles/external/uno.extensions/doc/Learn/Configuration/ConfigurationOverview.html).

**Reply by formalist** [reply]
Oh, I see. It worked running packaged. Thanks!
